test/specs/components/views/dropdown-menu-view-spec.js

```diff
@@ -0,0 +1,168 @@
+                keyPress = function( key ) {
+                    var event = document.createEvent('Event');
+
+                    event.keyCode = key;
+                    event.initEvent( 'keydown', true, false );
+                    document.dispatchEvent(event);
+                };
+
+            beforeEach(function() {
+                // Extend the view to add analytics for testing
+                var ExtendedDropdownMenuView = DropdownMenuView.extend({
+                    analyticsLinkClick: function( event ) {
+                        var $link = $(event.target),
+                            label = $link.hasClass('user-title') ? 'Dashboard' : $link.html().trim();
+
+                        /**
+                         *  Add your own analytics tracking here
+                         *  for example:
+                         */
+                        window.analytics.track( 'user_dropdown.clicked', {
+                            category: 'navigation',
+                            label: label,
+                            link: $link.attr('href')
+                        });
+                    }
+                });
+
+                // Set the DOM
+                setFixtures( '<div class="js-user-cta"></div>' );
+
+                dropdownModel.set({
+                    main: {
+                        text: 'username',
+                        url: 'dashboard'
+                    },
+                    button_label: 'User options dropdown',
+                    items: [
+                        {
+                            text: 'Dashboard',
+                            url: 'dashboard'
+                        }, {
+                            text: 'Account',
+                            url: 'account_settings'
+                        }, {
+                            text: 'Profile',
+                            url: 'learner_profile'
+                        }, {
+                            text: 'Sign Out',
+                            url: 'logout'
+                        }
+                    ]
+                });
+
+                view = new ExtendedDropdownMenuView({
+                    className: 'wrapper-more-actions user-menu logged-in',
+                    model: dropdownModel,
+                    parent: '.js-user-cta'
+                });
+
+                window.analytics = jasmine.createSpyObj('analytics', ['track', 'page', 'trackLink']);
+
+                jasmine.clock().install();
+            });
+
+            afterEach( function() {
+                view.remove();
+                jasmine.clock().uninstall();
+            });
+
+            it( 'should exist', function () {
+                expect( view ).toBeDefined();
+            });
+
+            it( 'should open the user menu on click of the button', function () {
+                var $btn = view.$el.find('.js-dropdown-button');
+
+                $btn.click();
+
+                setTimeout(function() {
```

Comment:
Some TravisCI race condition can occur so that the class is not added in time for the test to pass. This is to prevent this. I'm also not a fan of setTimeout but it is using the jasmine clock.